### PR TITLE
[tesseract] Add missing dependency pkgconf of feature training-tools

### LINKED ISSUE #42000: tesseract[training-tools] build failure because of missing dependency
### Operating system

Windows

### Compiler

MSVC

### Steps to reproduce the behavior

`vcpkg install tesseract[training-tools] --recurse`

### Failure logs

-- Fixing pkgconfig file: *VCPKG_DIR*/packages/tesseract_*TRIPLET*/debug/lib/pkgconfig/tesseract.pc
CMake Error at scripts/cmake/vcpkg_copy_tools.cmake:36 (message):
  Couldn't find tool "text2image":

      "VCPKG_DIR/packages/tesseract_TRIPLET/bin/text2image.exe" does not exist
Call Stack (most recent call first):
  ports/tesseract/portfile.cmake:58 (vcpkg_copy_tools)
  scripts/ports.cmake:192 (include)

### Additional context

If pkgconf package is already installed, the build will succeed.

### PR DESCRIPTION
**Pull request by LilyWangLL**
Fixes #42000
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [ ] ~~SHA512s are updated for each updated download.~~
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version.~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [ ] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/tesseract/vcpkg.json [hunk 1]
```diff
@@ -36,7 +36,11 @@
         "cairo",
         "fontconfig",
         "icu",
-        "pango"
+        "pango",
+        {
+          "name": "pkgconf",
+          "host": true
+        }
```
Comment (dg0yt) [inline ports/tesseract/vcpkg.json:43]: No, this is not how to vcpkg find acquire program PKGCONFIG.

**Review by Cheney-W — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thank you!

**Pull request merged**